### PR TITLE
lint: fix hack/tools/tools.go import issue

### LINKED ISSUE #308: [GoodFirst] fix hack/tools/tools.go gofmt issue
**What would you like to be added**:
`hack/tools/tools.go` not gofmted, (don't know why golangci-lint can't catch it).

We usually group import paths in the order of `std packages --> `third-party packages` --> `karmada packages`.
So the fix would be:
```diff
 import (
-       _ "k8s.io/code-generator"
        _ "github.com/onsi/ginkgo/ginkgo"
+       _ "k8s.io/code-generator"
 )
```

Please reply to this issue directly if you want to take this task. :)

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #308

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by daixiang0** [comment]
Need update to 1.40 to fix related issues.

**Comment by RainbowMango** [comment]
> Need update to 1.40 to fix related issues.

Why?

**Comment by daixiang0** [comment]
Some bugs, after update to latest version it works.

**Comment by kevin-wangzefeng** [comment]
It seems the lint is still failing. @daixiang0

**Comment by daixiang0** [comment]
Last version can find this issue with `build-tags` while fail in global mode, need time fixing.

**Comment by RainbowMango** [comment]
Hi, we have deprecated `golangci-lint-action`.

File: hack/verify-staticcheck.sh [hunk 1]
```diff
@@ -13,7 +13,18 @@ source "hack/util.sh"
...
+  local ret=0
+
+  golangci-lint run || ret=1
+
+  golangci-lint run --build-tags=tools ./hack || ret=1
+
+  return $ret
+}
+
+if [[ $ret -eq 0 ]] ; then
```
Comment (RainbowMango) [inline hack/verify-staticcheck.sh:27]: hack/verify-staticcheck.sh: line 27: ret: unbound variable

File: hack/verify-staticcheck.sh [hunk 2]
```diff
@@ -13,7 +13,18 @@ source "hack/util.sh"
...
 util::install_tools ${GOLANGCI_LINT_PKG} ${GOLANGCI_LINT_VER}
 
-if golangci-lint run; then
+
+function lint() {
+  local ret=0
+
+  golangci-lint run || ret=1
+
+  golangci-lint run --build-tags=tools ./hack || ret=1
```
Comment (RainbowMango) [inline hack/verify-staticcheck.sh:22]: `--build-tags strings             Build tags`
Can you explain what the `--build-tags` flags used for?  I'd prefer to add a comment on it.

Comment (daixiang0) [reply hack/verify-staticcheck.sh]: sure.

File: hack/verify-staticcheck.sh (hunk 1 above)
Comment (daixiang0) [reply hack/verify-staticcheck.sh]: fix.

File: hack/verify-staticcheck.sh [hunk 3]
```diff
@@ -13,7 +13,19 @@ source "hack/util.sh"
...
+
+  golangci-lint run || ret=1
+
+  # to test with specified build tags in hack, default skip them
+  golangci-lint run --build-tags=tools ./hack || ret=1
+
+  return $ret
+}
+
+if [[ lint -eq 0 ]] ; then
```
Comment (RainbowMango) [inline hack/verify-staticcheck.sh:28]: unbound variable

File: hack/verify-staticcheck.sh [hunk 4]
```diff
@@ -5,15 +5,26 @@ set -o nounset
...
-GOLANGCI_LINT_PKG="github.com/golangci/golangci-lint/cmd/golangci-lint"
-GOLANGCI_LINT_VER="v1.32.2"
+GOLANGCI_LINT_VER="v1.40.1"
 
 cd "${REPO_ROOT}"
 source "hack/util.sh"
 
-util::install_tools ${GOLANGCI_LINT_PKG} ${GOLANGCI_LINT_VER}
+# directly download binary, details refer to https://golangci-lint.run/usage/install
+curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin ${GOLANGCI_LINT_VER}
```
Comment (RainbowMango) [inline hack/verify-staticcheck.sh:14]: Why? For performance?

Comment (daixiang0) [reply hack/verify-staticcheck.sh:14]: Mostly, details see  https://golangci-lint.run/usage/install

Comment (RainbowMango) [reply hack/verify-staticcheck.sh:14]: Can we skip the download if the binary already exists?

Comment (daixiang0) [reply hack/verify-staticcheck.sh]: sure

**Comment by RainbowMango** [comment]
/close
Seems the issue is gone now. (Because we are using Go 1.16? not sure.)

**Comment by daixiang0** [comment]
May consider upgrade to 1.17 @RainbowMango

**Comment by RainbowMango** [comment]
> May consider upgrade to 1.17 @RainbowMango

We can do it when updating k/k to 1.22+

**Pull request closed without merging**